src/components/Button/index.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import styled from "@emotion/styled/macro";
+import LeafyGreenButton, { Variant } from "@leafygreen-ui/button";
+import { Icon } from "antd";
+
+interface ButtonType {
+  danger: string;
+  dark: string;
+  default: string;
+  info: string;
+  primary: string;
+}
+type ButtonTypeKeys = keyof ButtonType;
+
+interface Props {
+  loading?: boolean;
+  variant?: ButtonTypeKeys;
+  onClick?: () => void;
+  disabled?: boolean;
+  dataCy?: string;
+}
+
+export const Button: React.FC<Props> = ({
+  children,
+  loading,
+  variant = "default",
+  onClick,
+  disabled,
+  dataCy,
+}) => {
+  return (
+    <LeafyGreenButton
+      data-cy={dataCy}
+      variant={mapVariantToLeafyGreenVariant[variant]}
+      onClick={onClick}
+      disabled={disabled}
+    >
+      {loading && <StyledIcon type="loading" />}
```

Comment:
I think you can pass the Icon directly into the leafygreen button it has a `glyph` prop. 

https://github.com/mongodb/leafygreen-ui/blob/6b4cd8fecaa09350b739b9d3cf246a6ba30dfb3d/packages/button/src/Button.tsx#L298

Reply:
leafygreen does not have this icon -- it's from antd

Reply:
It takes a `ReactElement` as an input. So you should be able to pass `<StyledIcon/>` Directly into it.
https://github.com/mongodb/leafygreen-ui/blob/6b4cd8fecaa09350b739b9d3cf246a6ba30dfb3d/packages/button/src/Button.story.tsx#L39

Reply:
oh neat

Reply:
haha i had to do this exact thing like 100 times during my stitch rotation.